test/immutable-store-state.js

```diff
@@ -0,0 +1,40 @@
+import Alt from '../dist/alt-with-runtime'
+import Immutable from 'immutable'
+import { assert } from 'chai'
+import sinon from 'sinon'
+
+const alt = new Alt()
+
+const actions = alt.generateActions('fire')
+
+const assign = require('object-assign')
+
+const store = alt.createStore({
+  displayName: 'ImmutableStore',
+
+  state: Immutable.Map({}),
```

Comment:
The thing with `Record`s is that they do not have any setters, it just throws an error, so I am not sure how useful it would be in a store, where you presumably would want to change your data on actions.

`Records` are nice because you can get the values from them without an explicit setter `obj.val`, but they are immutable.  So they are nice if you wrap mutable data in your store and return them as a `Record` such that they are immutable in the views, but I don't think they make sense in this context.

Maybe a `List`?  I think if we could make this configurable, it would be nice so we don't have to choose `Map` or `List`.  We could default to `Map` because that seems sensible, but provide an escape hatch to configure?